tools/ci-skip-for-maintainers.js

```diff
@@ -12,90 +12,82 @@ const MAINTAINERS_REGEX = new RegExp(
   "g",
 );
 
-const main = async () => {
-  const markdownContent = readFileSync(MaintainersFile, "utf-8");
+const getMaintainersFileContent = () => readFileSync(MaintainersFile, "utf-8");
 
-  const args = process.argv.slice(2);
-  const pullReqUrl = args[0];
-  const committerLogin = args[1];
+// Function to get the latest commit message author
+const getCommitterLogin = () => {
+  const authorBuffer = execSync("git log -1 | grep Author | cut -d' ' -f2");
+  const authorString = authorBuffer.toString();
+  const authorStringTrim = authorString.trim();
+  return authorStringTrim;
+};
 
-  //Uncomment these lines and change it for local machine testing purposes:
-  //const pullReqUrl = "https://api.github.com/repos/<username>/cactus/pulls/<number>";
-  //const committerLogin = "<username>";
+// Function to get the latest commit message
+const getLatestCommitMessage = () => {
+  const commitMsgBuffer = execSync("git log -1 --pretty=%B");
```

Comment:
@petermetz to answer your question, this `git log` shell command is the alternative we used for fetching commit messages instead of calling github API to avoid limits and crashing

Reply:
@zondervancalvez I'll post a screenshot here with the question(s). Maybe you posted the answers somewhere else and I just couldn't find it? Pull request comments/reviews get confusing quick.

![image](https://github.com/user-attachments/assets/7bb0d02f-74e1-4cb9-85af-af340e00f9aa)

Reply:
@petermetz I added `if: always()` on the jobs that requires `check-ci-skip` job. This will bypass check-ci-skip job if the job failed/crashes.

Reply:
@zondervancalvez OK, but would that then mean that the skip CI marker does not work anymore?

What I mean is that the .js script currently exists with a job failure when it is trying to indicate that the CI should be skipped. But with `if: always()` the jobs will ignore that. If I'm understanding it correctly.

So I'm thinking you need to refactor the .js script to use a different signaling mechanism where it succeeds as long as there was no crash in it, and it places the information about the CI being skipped  or not into some other context variable  (which must not be writeable by pull request code because then people could just hack it by writing to this place where we store the information about the CI being skipped or not)

```js
if (isMaintainer) {
  console.log(
    "CI skipped as per request. Exit with an error to PAUSE dependent workflows.",
  );
  process.exit(1);
} else {
  process.exit(0);
}
```